From b9d983f54ddddff1b7570ea709f904c4fbc2165f Mon Sep 17 00:00:00 2001
From: Matt Gumbel <matthew.k.gumbel@intel.com>
Date: Thu, 2 May 2013 17:37:26 -0700
Subject: bt_hci_bdroid: check for failure to open hci

Initialization should not continue if we failed to open the hci transport.

Change-Id: Idf2008477f34ed216163ca47be637f415d640bf5
Signed-off-by: Matt Gumbel <matthew.k.gumbel@intel.com>
---
 hci/src/bt_hci_bdroid.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/hci/src/bt_hci_bdroid.c b/hci/src/bt_hci_bdroid.c
index 8e4f8ff..a184fa8 100644
--- a/hci/src/bt_hci_bdroid.c
+++ b/hci/src/bt_hci_bdroid.c
@@ -461,7 +461,8 @@ static void *bt_hc_worker_thread(void *arg)
 
         if (events & HC_EVENT_PRELOAD)
         {
-            userial_open(USERIAL_PORT_1);
+            if (userial_open(USERIAL_PORT_1) == FALSE)
+                break;
 
             /* Calling vendor-specific part */
             if (bt_vnd_if)
-- 
1.7.4.1

